### ISSUE TITLE
Compiler warnings on Linux

### ISSUE THREAD

**Issue by whipowill** [ZhengPeiRu21/mod-playerbots#61]
Probably 99% of all the warnings on the compile are this:

```
/home/azeroth/core/src/server/game/World/World.h:344:31: warning: 'GetPlayerbotsDBRevision' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
```

**Comment by ZhengPeiRu21** [comment]
Thank you for attention to this issue! Would you please try after this commit to AzerothCore branch (https://github.com/ZhengPeiRu21/azerothcore-wotlk/commit/9a6c789f1a52fa43e3bb58420980f6841ad6506b) to see if this issue is resolved?

**Comment by whipowill** [comment]
Some things:

1)  When I compile ``azerothcore-wotlk branch playerbots`` without module ``mod-playerbots`` I get this error on server load:

```
Initialize data stores...
[1146] Table 'acore_world.charsections_dbc' doesn't exist
```

2) When I compile ``azerothcore-wotlk branch playerbots`` with module ``mod-playerbots`` it works, but when I load the server I get the error I mentioned here: https://github.com/ZhengPeiRu21/mod-playerbots/issues/59

3) When I compile your ``azerothcore-wotlk branch playerbots`` with module ``mod-playerbots`` I get a lot of warnings.  I was only able to capture the last 2000 lines of the Tmux output, which I have attached for your review: [output.txt](https://github.com/ZhengPeiRu21/mod-playerbots/files/8709703/output.txt)

**Comment by Ramys** [comment]
https://github.com/ZhengPeiRu21/azerothcore-wotlk/tree/Playerbot

**Comment by ZhengPeiRu21** [comment]
Thank you again for the reports. Because LOAD_DBC tables are given within a define statement, it is not possible to use ifndef to check if MOD_PLAYERBOTS is available, so the table does need to be present to run server compiled from the branch. The correct solution I feel is probably to move playerbot table data to a separate LOAD_DBC, but support for running a not Playerbots with the Playerbots branch is right now a less priority than fixing some other issues.

**Comment by whipowill** [comment]
I started working on squishing all the compile bugs:
https://github.com/whipowill/wow-module-playerbots/tree/compilebugs

I'm trying to fix two dozen or so at a time and recompile to see what happens.

I don't know C++ so this may end up being a fools errand, but I'm trying to help out.  If I'm successful I will submit a pull request.

**Comment by whipowill** [comment]
My fixes seem to be working.  I'm still going thru the compile output looking for ones I can fix with my limited knowledge.  The ones I fixed are the ones about "unused" arguments on methods (which I fixed by adding ``[[maybe_unused]]``), and method arguments being passed in the wrong order.  

The rest of these warnings I don't know how to fix, but I will post my latest terminal output for you to review when this compile finishes.  Most of them have to do with "destructors" and how you can't delete things improperly.

Let me know if you like my fixes and if you want them as a pull request.  I initialed every line where I made a change so they'd be easy to find if I screwed something up.

**Comment by whipowill** [comment]
Latest output
[output.txt](https://github.com/ZhengPeiRu21/mod-playerbots/files/8730141/output.txt)

**Comment by ZhengPeiRu21** [comment]
Yes, please submit a PR with your fixes! I am compiling with MSVC on Windows and it seems for more lenient with warnings, so I do not see many of these otherwise. Thank you for continued efforts to improve this module!

**Comment by whipowill** [comment]
Ya, np I'm pumped to get this working and I hope you continue to work on it.  I'm running this on a remote Ubuntu server and the compile takes 30 minutes, so testing changes on this thing is going to be a doozy.

**Comment by whipowill** [comment]
It sounds like people on Windows have fewer or no compile warnings, but people on linux have a ton.  I just changed the compile on my linux setup from clang to gcc and I'm hoping to see if I have fewer compile warnings.  My thinking right now is that you and I have different experiences with these bots -- where they probably work a lot better for you on Windows than they do for me on linux, and it's because of these compile issues.

**Comment by ZhengPeiRu21** [comment]
I apologize for slow responding. I earlier did not see your reply. I believe bot behavior is identical on Windows and Linux (which is meaning still needing much improvement, hehe), but compiler is just more noisy on Linux.
